.github/workflows/ci-tests.yml

```diff
@@ -0,0 +1,70 @@
+name: "ci tests"
+
+on:
+  push:
+    branches:
+      - main
+      - plain-cumulus-template
+    paths-ignore:
+      - "**.md"
+  pull_request:
+    paths-ignore:
+      - "**.md"
+  workflow_dispatch:
+    inputs:
+      test-macos-and-windows:
+        description: "run macOS and Windows tests"
+        required: true
+        default: false
+        type: boolean
+
+env:
+  # Not needed in CI, should make things a bit faster
+  CARGO_INCREMENTAL: 0
+  CARGO_TERM_COLOR: always
+  # Remove unnecessary WASM build artefacts
+  WASM_BUILD_CLEAN_TARGET: 1
+
+jobs:
+  clippy-fmt-test:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-22.04
+          - macos-12
+          - windows-2022
+        run-all:
+          - ${{ inputs.test-macos-and-windows == true || github.ref == 'refs/heads/main' }}
+        exclude: # exclude macos-12 and windows-2022 when the condition is false
+          - run-all: false
+            os: macos-12
+          - run-all: false
+            os: windows-2022
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: git checkout
+        uses: actions/checkout@v4
+
+      - name: Install Protoc
+        uses: arduino/setup-protoc@9b1ee5b22b0a3f1feb8c2ff99b32c89b3c3191e9 # v2.0.0
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: install rust toolchain
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+
+      - name: cargo fmt
```

Comment:
let me adjust the commands for release

Reply:
clippy already includes check, so I made both clippy and test command use the release. 

For build, I think we don't need it. When we have a release workflow (if we ever have), then we can utilize build there. 

Afaik build won't provide additional checks for us on top of `clippy` and `test`

Reply:
- for cargo check: clippy itself runs cargo check: https://github.com/rust-lang/rust-clippy/blob/c154754b74577906c5d55d57f7daeff02d6a33e7/src/main.rs#L59
- for `run`, I don't see how that would be useful for the CI workflow
- for `build`, I may be wrong, so if you know any possibility that `build` may fail whilst `clippy` and `test` are succeeding, we can for sure add `build` command to this one as well. 

My point was that if we plan to have a release workflow (I didn't say we shouldn't :D it was just not discussed yet), we will already include the `build` command there. 

Compiling substrate code and running multiple commands on it already makes the workflow heavy, I suggest we keep it as minimal as we can.

Reply:
> usually there is one command for cargo check --release and separate commands for clippy and test because both of those compile in std and the cargo build/check/run --release compiles in no_std to separately test wasm compilation

ah ok, didn't know that, thanks!